edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/pvc/KubernetesPvcMapper.cs

```diff
@@ -67,7 +67,16 @@ V1PersistentVolumeClaim ExtractPvc(KubernetesModule module, Mount mount, IDictio
                 persistentVolumeClaimSpec.StorageClassName = this.storageClassName.OrDefault();
             }
 
-            return new V1PersistentVolumeClaim(metadata: new V1ObjectMeta(name: volumeName, labels: labels), spec: persistentVolumeClaimSpec);
+            var ownerReferences = new List<V1OwnerReference>
```

Comment:
Like the caller of this class creates the list once?

Reply:
Good point about redundant code (something I will task myself to refactor in near future). Unfortunately, the K8s object creation code in Agent is at the end of the call stack. And we are forced to recreate the same metadata objects for each K8s object type, even though they all share the same labels, owners, etc.

Reply:
Agreed about not taking too much time, but you could very quickly write a function or an extension method to at least not contribute to the redundant code...

Reply:
Makes sense. It isn't an extension method however. It will just be within a separate utility class that does the one way conversion from KubernetesModuleOwner->V1ObjectReference.